Formula/kt-connect.rb

```diff
@@ -0,0 +1,30 @@
+class KtConnect < Formula
+  desc "Toolkit for integrating with kubernetes dev environment more efficiently"
+  homepage "https://alibaba.github.io/kt-connect/#/"
+  url "https://github.com/alibaba/kt-connect/archive/refs/tags/v0.3.5.tar.gz"
+  sha256 "b1a2f0f71feb7d8c4418f73048c24687d513dc1cabb1f68ffac3be5baa0e3f5a"
+  license "GPL-3.0-or-later"
+
+  depends_on "go" => :build
+
+  def install
+    ldflags = "-s -w -X main.version=#{version}"
+    system "go", "build", *std_go_args(ldflags: ldflags), "-o", bin/"ktctl", "./cmd/ktctl"
+
+    # Install bash completion
+    output = Utils.safe_popen_read(bin/"ktctl", "completion", "bash")
+    (bash_completion/"ktctl").write output
+
+    # Install zsh completion
+    output = Utils.safe_popen_read(bin/"ktctl", "completion", "zsh")
+    (zsh_completion/"_ktctl").write output
+
+    # Install fish completion
+    output = Utils.safe_popen_read(bin/"ktctl", "completion", "fish")
+    (fish_completion/"ktctl.fish").write output
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/ktctl --version")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.